pkg/service/configuration_manager.go

```diff
@@ -15,38 +16,88 @@ type ConfigurationManager interface {
 	WriteConfiguration(config *model.Config) error
 }
 
-func NewConfigurationManager(configFile string, remote bool) (ConfigurationManager, error) {
-	uri, err := url.Parse(configFile)
+type Downloader interface {
```

Comment:
`Reader` with a `Read` method is preferable in this context, since you don't download a local file.